### PR TITLE
Fix backward weight igemm kernel time bug

### PR DESCRIPTION
**Pull request by TejashShah**
It fixes the bug reported in #199

### CODE REVIEW

File: src/ocl/convolutionocl.cpp [hunk 1]
```diff
@@ -4187,6 +4187,8 @@ void ConvolutionDescriptor::ConvolutionBackwardWeights(Handle& handle,
         break;
         case miopenConvolutionBwdWeightsAlgoImplicitGEMM:
         {
+            float elapsed = 0.0;
+            handle.ResetKernelTime();
```
Comment (JehandadKhan) [inline src/ocl/convolutionocl.cpp:4191]: If you are doing the accumulation yourself, then this is redundant

Comment (TejashShah) [reply src/ocl/convolutionocl.cpp]: Thanks. Done.

**Comment by daniellowell** [comment]
@TejashShah Formatting.

**Comment by atamazov** [comment]
The time is measured incorrectly during the "Run" phase, not during "Find". So it does not affect results of Find, find-db, and performance. I am in doubt if this PR is a must for ROCm 3.5.

**Comment by TejashShah** [comment]
> 
> 
> The time is measured incorrectly during the "Run" phase, not during "Find". So it does not affect results of Find, find-db, and performance. I am in doubt if this PR is a must for ROCm 3.5.

Do you mean, in general, the time reported along with this str: GPU Kernel Time Backward Weights
is not reliable and shouldn't be used for benchmarking?

**Comment by TejashShah** [comment]
@atamazov I think, it is a must

As the time reported for bfp16/fp16 group config is way off:

MIOpenDriver convfp16 -n 128 -V 0 -c 1024 -H 14 -W 14 -k 1024 -y 3 -x 3 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -g 32 -F 4 -t 1 -i 1
MIOpen(HIP): Info [FindConvBwdWeightsAlgorithm] BWrW Chosen Algorithm: ConvHipImplicitGemmV4R4GenWrWXdlops , 1179648, 4.35072
GPU Kernel Time Backward Weights Conv. Elapsed: 0.022720 ms (average)


MIOpenDriver convbfp16 -n 128 -V 0 -c 1024 -H 14 -W 14 -k 1024 -y 3 -x 3 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -g 32 -F 4 -t 1 -i 1
MIOpen(HIP): Info [FindConvBwdWeightsAlgorithm] BWrW Chosen Algorithm: ConvHipImplicitGemmV4R4GenWrWXdlops , 1179648, 6.80769
GPU Kernel Time Backward Weights Conv. Elapsed: 0.024000 ms (average)

Another reason the fix is required is that fp16/bfp16 backward wrw igemm kernels invokes cast kernel at the end. If we don;t gather time like in this PR, in group configs, it only reports cast kernel time.

**Comment by atamazov** [comment]
@TejashShah 
> Do you mean, in general, the time reported along with this str: GPU Kernel Time Backward Weights is not reliable and shouldn't be used for benchmarking?

No. This time should be reliable. We can use it for our internal purposes (internal testing , triaging...), and this issue is a bug, definitely.

However the issue does NOT affect results of Find, find-db, and actual performance of the library.

**Pull request merged**